File: src/main/java/com/szokone/aem/gokotlin/java/model/SampleModel.java
package com.szokone.aem.gokotlin.java.model;

import com.szokone.aem.gokotlin.java.service.ConfigurableService;
import org.apache.sling.api.resource.Resource;
import org.apache.sling.models.annotations.DefaultInjectionStrategy;
import org.apache.sling.models.annotations.Model;

import javax.annotation.PostConstruct;
import javax.inject.Inject;

@Model(adaptables = Resource.class, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
public class SampleModel {

    @Inject
    private String textfield;

    @Inject
    private Long numberfield;

    @Inject
    private boolean checkbox;

    private String calculatedValue;

    @Inject
    private ConfigurableService service;

    @PostConstruct
    public void init() {
        calculatedValue = service.getReplacements().toString() + " | " + textfield + " | " + numberfield + " | " + checkbox;
    }

    public String getTextfield() {
        return textfield;
    }

    public Long getNumberfield() {
        return numberfield;
    }

    public boolean isCheckbox() {
        return checkbox;
    }

    public String getCalculatedValue() {
        return calculatedValue;
    }
}

File: src/main/kotlin/com/szokone/aem/gokotlin/kotlin/model/SampleModel.kt
package com.szokone.aem.gokotlin.kotlin.model

import com.szokone.aem.gokotlin.kotlin.service.ConfigurableService
import org.apache.sling.api.resource.Resource
import org.apache.sling.models.annotations.DefaultInjectionStrategy
import org.apache.sling.models.annotations.Model
import javax.annotation.PostConstruct
import javax.inject.Inject
import javax.inject.Named

@Model(adaptables = [Resource::class], defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
class SampleModel {

    // NOTE: Use this approach when injecting objects like String...
    @Inject
    lateinit var textfield: String

    // NOTE: Use this approach when injecting primitives like Long, Boolean...
    @JvmField
    @field:[Inject Named("numberfield")]
    var numberfield: Long = 0

    @JvmField
    @field:[Inject Named("checkbox")]
    var checkbox: Boolean = false

    lateinit var calculatedValue: String

    @Inject
    lateinit var service: ConfigurableService

    @PostConstruct
    fun init() {
        calculatedValue = "${service.getReplacements()} | ${textfield} | ${numberfield} | ${checkbox}"
    }
}